pages/data-providers/[data-provider-id]/settings.jsx

```diff
@@ -1,84 +1,103 @@
-import React, { useReducer, useCallback, useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { classNames } from '@oacore/design/lib/utils'
 
 import styles from './settings.css'
 
 import { Card, TextField, Button } from 'design'
 import { withGlobalStore } from 'store'
 
-const Settings = ({ store, className, ...restProps }) => {
-  const [isSaving, setIsSaving] = useState(false)
-  const [hasFormChanged, setHasFormChanged] = useState(false)
-
-  const reducer = (state, { name, value }) => {
-    setHasFormChanged(true)
-    return {
-      ...state,
-      [name]: value,
-    }
+const SettignsGroup = ({
+  children,
+  className,
+  tag,
+  onSubmit,
+  ...formProps
+}) => {
+  const [isChanged, setChanged] = useState(false)
+  const [isSubmited, setSubmitted] = useState(false)
+
+  const handleChange = () => setChanged(true)
+
+  const handleSubmit = event => {
+    setSubmitted(true)
+    if (onSubmit) onSubmit(event)
   }
 
-  const [formState, dispatch] = useReducer(reducer, {
-    name: store.dataProvider.name,
-    institution: store.dataProvider.institution,
-    email: store.dataProvider.email,
-  })
+  useEffect(() => {
+    setChanged(false)
+    setSubmitted(false)
+  }, [children])
+
+  return (
+    <Card className={classNames.use(styles.section).join(className)} tag={tag}>
+      <form
+        method="post"
+        onChange={handleChange}
+        onSubmit={handleSubmit}
+        {...formProps}
+      >
+        {children}
+        {isChanged && (
+          <Button variant="contained" disabled={isSubmited}>
+            Save
+          </Button>
+        )}
+      </form>
+    </Card>
+  )
+}
 
-  const handleChange = useCallback(e => {
-    dispatch({ name: e.target.name, value: e.target.value })
-  }, [])
+const Settings = ({ store, className, ...restProps }) => {
+  const { dataProvider } = store
+  const organization = { name: dataProvider.institution }
 
-  const handleSubmit = useCallback(async e => {
-    e.preventDefault()
+  const handleSubmit = event => {
+    event.preventDefault()
 
-    // TODO: API call here
-    try {
-      setIsSaving(true)
-      await store.changeDataProviderMetadata(formState)
-    } finally {
-      setIsSaving(false)
-    }
-  }, [])
+    const formData = new FormData(event.target)
+    const data = Object.fromEntries(formData.entries())
+
+    const scope = event.target.getAttribute('name')
+    const present = {
+      organization: store.updateOrganization,
```

Comment:
```suggestion
      'organization': store.updateOrganization,
```

Reply:
Yes, it should be so. However, the branch is not updated with master and the current branch  setup is what you see above.